ClassRegistration/ClassRegistration.App/Controllers/SectionController.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using ClassRegistration.DataAccess.Entity;
+using ClassRegistration.Domain.Repositories;
+using ClassRegistration.Domain.Interfaces;
+
+
+
+namespace ClassRegistration.App.Controllers
+{
+
+    [Route("api/[controller]")]
+    public class SectionController : ControllerBase
+    {
+
+        private readonly ISectionRepository _secRepo;
+        
+
+        public SectionController(ISectionRepository secRepo)
+        {
+            _secRepo = secRepo;
+           
+        }
+
+
+
+        // GET api/section/items/50
+        [HttpGet("items/{id}")]
+        public ActionResult<Section> GetCourseByInstructorID(int id)
+        {
+            //get all the sections and associated courses for an instructor
+            var theSections = _secRepo.GetSectionByInstID(id);
+            
+
+
+            if (!theSections.Any())
+            {
+                return NotFound();
+            }
+            return Ok(theSections);
+
+
+
+        }
+
+
+        
+
+
+
+        //get all the sections available
+        // GET: api/section/items
+        [HttpGet("items")]
+        public IActionResult GetAllSectionsAvailable()
```

Comment:
Since we're getting all sections and the base URL is already `/api/section`, we can just leave this as the default `Get` method setup and pattern (i.e., just change the function name back to `Get` and leave the annotation as `[HttpGet]`)

Reply:
I just wrote this method to test the navigation properties were correct in postman. I added a comment about this in the new commit. I don't think we'd ever need to display all the sections without having the associated courses along with that.